lib/library.hoon

```diff
@@ -0,0 +1,135 @@
+/-  *post, resource, library, spider
+/+  *graph-store, strandio
+|%
+++  incr-index
+  :: increments the last value in the index by 1
+  |=  [=index:post]
+  ^-  index:post
+  =/  last=@                (sub (lent index) 1)  :: calculate the array index of the last item
+  =/  old-revision-count=@  (snag last index)     :: get the last item
```

Comment:
Can use `+rear` instead.

Reply:
So I could replace `(snag last index)` with `(rear index)`, but in the call to `+snap` I'd still need `last`, right? Would you know of a better way to replace the tail of the index so we could get rid of it altogether?

Reply:
`(snoc (snip index) +((rear index)))`

Reply:
https://github.com/ynx0/library/pull/3/commits/3035efda739f00f164a40ec78364fc16101b8913